Cargo.toml

```diff
@@ -48,17 +48,22 @@ usbd-serial = "0.1.0"
 [features]
 default = ["unproven"]
 device-selected = []
+needs-subvariant = []
 rt = ["stm32f3/rt"]
-stm32f301 = ["stm32f3/stm32f301", "device-selected"]
+stm32f301 = ["stm32f3/stm32f301", "needs-subvariant"]
```

Comment:
Using the deprecated attribute seems to be impossible for features,

`needs-subvariant` allows for a kind of _deprecate compiler error_:
`device-selected` is only the opposide of `needs-subvariant` if the feature starts with `stm32f`. 
Using `needs-subvariant` allows to discriminate between a user not going into detail and a user not setting `stm32fxxx` at all.

Reply:
That's true. But to make it effective as a deprecation, I would not change it to a compile error. 

Either keep the `needs-subvariant` feature, and spit out a less verbose warning (does `compiler_warn!` exist?), which states that the this feature is deprecated and choose a Variant, which has the least common features (e.g. `stm32f303x6` for the f303s). 
We could keep it that way for one or more version and than remove the intermediate step (soft breaking change)

Or remove the `needs-subvariant` feature (but this is not my favored way), which results into a compile error of the feature is still selected by the user (this would be a hard breaking change)

Either way, we should put a warning / breaking change note in the changelog.

Reply:
> Either keep the needs-subvariant feature, and spit out a less verbose warning (does compiler_warn! exist?),

Unfortunately, `compier_warn!` doesnt exist (see discussion [here](https://internals.rust-lang.org/t/pre-rfc-add-compile-warning-macro/9370) and links there) and I couldnt find a similar way of warning the user.

> choose a Variant, which has the least common features

Actually `stm32f303` would be the one with the least common features. Thats where all the confusion arises. So thats why I'm very much in favor of incrementing the version number.

Reply:
Argh. So nothing we can do to cleanly deprecate this :/ 

So then, we should keep the `needs_subvariant` feature :)